File: qt/properties/src/MainWindow.cpp
#include "MainWindow.h"
#include "ui_MainWindow.h"

#include <iostream>

MainWindow::MainWindow(QWidget *parent) :
    QMainWindow(parent),
    ui(new Ui::MainWindow)
{
    ui->setupUi(this);

    connect( this, SIGNAL(varChanged(int)), this, SLOT(onVar_changed(int)) );
}

MainWindow::~MainWindow()
{
    delete ui;
}

void MainWindow::on_pushButton_clicked()
{
    setVar( 2 );
    std::cout << _var << std::endl;
}

void MainWindow::onVar_changed(int var)
{
    std::cout << "var= " << var << std::endl;
}

File: python/PyQt/04_signals_and_slots/MainWindow.py
from PyQt5.QtWidgets import *

from Ui_MainWindow import Ui_MainWindow

class MainWindow(QMainWindow, Ui_MainWindow):
    def __init__(self, *args, obj=None, **kwargs):
        super(MainWindow, self).__init__(*args, **kwargs)
        self.setupUi(self)
        self.setWindowTitle("My First Qt App")
        # self -> object -> signal -> connect -> slot
        self.pushButton_2.clicked.connect(self.changeLabelTextButton2Clicked)
    

    def on_pushButton_1_clicked(self):
        self.label.setText("Button 1 clicked")
    

    def changeLabelTextButton2Clicked(self):
        self.label.setText("Button 2 clicked")